chapter_2/component_interface/CMakeLists.txt

```diff
@@ -3,10 +3,26 @@
 # SPDX-License-Identifier: MIT
 
 project(
-  ch2.framework.component_interface
-  VERSION 1.0
-  DESCRIPTION "Chapter 2 component interface implementation"
-  LANGUAGES CXX)
+    ch2.framework.component_interface
+    VERSION 1.0
+    DESCRIPTION "Chapter 2 component interface implementation"
+    LANGUAGES CXX
+)
 
 add_library(${PROJECT_NAME} INTERFACE)
-target_include_directories(${PROJECT_NAME} INTERFACE include/)
+target_include_directories(
+    ${PROJECT_NAME} INTERFACE include/ $<INSTALL_INTERFACE:include>
```

Comment:
I usually specify the build_interface explicitely.

```
target_include_directories(${PROJECT_NAME}
   INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}
```

Reply:
The `BUILD_INTERFACE` genex will be useful later when you reach the topic of relocatable CMake packages.

Reply:
Removed this.